src/Notification.vala

```diff
@@ -162,6 +146,29 @@ public class Notifications.Notification : GLib.Object {
         }
     }
 
+    private GLib.Icon find_icon () {
+        var icon_theme = Gtk.IconTheme.get_for_display (Gdk.Display.get_default ());
+
+        // Always "" if sent by GLib.Notification
+        if (app_icon == "" && app_info != null && icon_theme.has_gicon (app_info.get_icon ())) {
+            return app_info.get_icon ();
+        }
+
+        if (app_icon.contains ("/")) {
+            var file = File.new_for_uri (app_icon);
+            if (file.query_exists ()) {
```

Comment:
Not related to the linked issue, but I guess we can use `GLib.FileUtils.test (app_icon, IS_REGULAR)` instead of creating a File object that actually we don't need?

https://valadoc.org/glib-2.0/GLib.FileUtils.test.html